### PR TITLE
Added some new compiler Option(Warning flags).

### PR DESCRIPTION
**Pull request by monadbobo**
1. Added some new compiler Option(Warning flags).
2. fixed the compile warning.

### CODE REVIEW

**Comment by monadbobo** [comment]
Jenkins Go!

**Comment by dangleptr** [comment]
It's a good improvement.  I think we could take it in release 1.x.  
@monadbobo Could you rebase it on current master?

File: src/storage/StorageServer.cpp [hunk 1]
```diff
@@ -59,9 +59,9 @@ std::unique_ptr<kvstore::KVStore> StorageServer::getStoreInstance() {
         }
         return nbStore;
     } else if (FLAGS_store_type == "hbase") {
-        LOG(FATAL) << "HBase store has not been implemented";
```
Comment (Shylock-Hg) [inline src/storage/StorageServer.cpp:62]: Why change this ?

**Comment by CPWstatic** [comment]
The compiling part had been refactored since nebula 2.0.

**Pull request closed without merging**